packages/alchemy/src/react/hooks/useAddPasskey.ts

```diff
@@ -0,0 +1,53 @@
+"use client";
+
+import { useMutation, type UseMutateFunction } from "@tanstack/react-query";
+import { useAlchemyAccountContext } from "../context.js";
+import type { BaseHookMutationArgs } from "../types.js";
+import { useSigner } from "./useSigner.js";
+
+export type UseAddPasskeyData = string[];
+
+export type UseAddPasskeyParams = CredentialCreationOptions | undefined | void;
+
+export type UseAddPasskeyMutationArgs = BaseHookMutationArgs<
+  UseAddPasskeyData,
+  UseAddPasskeyParams
+>;
+
+export type UseAddPasskeyResult = {
+  addPasskey: UseMutateFunction<
+    UseAddPasskeyData,
+    Error,
+    UseAddPasskeyParams,
+    unknown
+  >;
+  isAddingPasskey: boolean;
+  error: Error | null;
+};
+
+export function useAddPasskey(
+  mutationArgs?: UseAddPasskeyMutationArgs
+): UseAddPasskeyResult {
+  const { queryClient } = useAlchemyAccountContext();
+  const signer = useSigner();
+
+  const {
+    mutate: addPasskey,
+    isPending: isAddingPasskey,
+    error,
+  } = useMutation(
+    {
+      mutationFn: async (params: UseAddPasskeyParams) => {
+        return signer!.addPasskey(params ?? undefined);
```

Comment:
for all out hooks, we should check from AlchemyAccountContext which state the user/account is in, and throw informative error messages instead of just throwing null exceptions or other generic error messages

Reply:
the useSigner hook will throw an error if there is no clientStore we can get the signer from (and thus the user, account, etc.).

let's chat more with Moldy when he's back. he actually advised [here](https://github.com/alchemyplatform/aa-sdk/pull/566#discussion_r1561263929) that i move this approach.